addons/hr_expense/models/hr_expense_sheet.py

```diff
@@ -736,14 +770,19 @@ def _prepare_bills_vals(self):
 
     def _prepare_move_vals(self):
         self.ensure_one()
-        return {
-            # force the name to the default value, to avoid an eventual 'default_name' in the context
-            # to set it to '' which cause no number to be given to the account.move when posted.
+        to_return = {
             'name': '/',
-            'date': self.accounting_date or max(self.expense_line_ids.mapped('date')) or fields.Date.context_today(self),
             'expense_sheet_id': self.id,
         }
 
+        if self.payment_mode == 'company_account':
+            to_return['date'] = max(self.expense_line_ids.mapped('date')) or fields.Date.context_today(self)
+            to_return['invoice_date'] = fields.Date.context_today(self)
+        else:
+            to_return['invoice_date'] = self.accounting_date or self._compute_default_accounting_date(max(self.expense_line_ids.mapped('date')))
```

Comment:
As `accounting_date` is a compute, wouldn't it always be set?
Is there a case in which the second part gets called?